src/js/components/Marathon.jsx

```diff
@@ -66,28 +56,70 @@ var Marathon = React.createClass({
     }.bind(this));
 
     Mousetrap.bind("c", function () {
-      router.navigate("newapp", {trigger: true});
+      router.transitionTo("newapp");
     }, "keyup");
 
     Mousetrap.bind("g a", function () {
       if (this.state.modalClass == null) {
-        router.navigate("apps", {trigger: true});
+        router.transitionTo("apps");
       }
     }.bind(this));
 
     Mousetrap.bind("g d", function () {
       if (this.state.modalClass == null) {
-        router.navigate("deployments", {trigger: true});
+        router.transitionTo("deployments");
       }
     }.bind(this));
 
     Mousetrap.bind("shift+,", function () {
-      router.navigate("about", {trigger: true});
+      router.transitionTo("about");
     });
 
     this.startPolling();
   },
 
+  componentWillReceiveProps: function () {
+    this.onRouteChange();
+  },
+
+  onRouteChange: function () {
+    var router = this.context.router;
+
+    var params = router.getCurrentParams();
+    var path = router.getCurrentPathname();
+    var modalQuery = router.getCurrentQuery().modal;
+    var modalClass = null;
+
+    if (modalQuery === "newapp") {
+      modalClass = NewAppModalComponent;
+    } else if (modalQuery === "about") {
+      modalClass = AboutModalComponent;
+    }
+
+    var appId = params.appid != null ?
+      decodeURIComponent(params.appid) :
+      params.appid;
```

Comment:
We haven't even agreed on using ternaries...can we discuss first, before introducing new styles into the code?